logging/logging_test.go

```diff
@@ -1358,9 +1355,90 @@ func TestRedirectOutputFormats(t *testing.T) {
 	}
 }
 
-func ExampleRedirectAsJson_withStdout() {
-	// use previously created client
+func TestInstrumentationIngestion(t *testing.T) {
+	var got []*logpb.LogEntry
+
+	tests := []struct {
+		name          string
+		ingestionFlag bool
+		want          int
+	}{
+		{
+			name:          "first time log",
+			ingestionFlag: true,
+			want:          2,
+		},
+		{
+			name:          "any non first log",
+			ingestionFlag: false,
+			want:          1,
+		},
+	}
+	// setup fake client
+	client, err := fakeClient("projects/test", func(e *logpb.WriteLogEntriesRequest) {
+		got = e.GetEntries()
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+	iiStatus := internal.IngestInstrumentation()
+
+	entry := &logging.Entry{Severity: logging.Info, Payload: "test string"}
+	logger := client.Logger("test-instrumentation")
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got = nil
+			internal.SetIngestInstrumentation(tc.ingestionFlag)
+			err := logger.LogSync(context.TODO(), *entry)
```

Comment:
nit: TODO implys there will be eventual code changes here. I would just make a context.Background at the top of this test and pass that to all calls.